Quadriga/src/main/java/edu/asu/spring/quadriga/web/network/NetworkListController.java

```diff
@@ -15,87 +15,150 @@
+     * 
+     * @author Lohith Dwaraka, Chiraag Subramanian
+     * @param networkId
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws JAXBException
+     */
+    @RequestMapping(value = "auth/networks/visualize/{networkId}", method = RequestMethod.GET)
+    public String visualizeNetworks(@PathVariable("networkId") String networkId, ModelMap model, Principal principal)
+            throws QuadrigaStorageException, JAXBException {
+        // Identify the User
+        String userId = principal.getName();
+
+        // Get network details using the networkId
+        INetwork network = networkManager.getNetwork(networkId);
+
+        // Get details of workspace associated with the network using
+        // network-workspace mapping details
+        IWorkSpace workspace = network.getNetworkWorkspace().getWorkspace();
+
+        // Get projectId of the project to which the workspace belongs
+        String projectId = workspaceManager.getProjectIdFromWorkspaceId(workspace.getWorkspaceId());
+
+        // Get project details using projectId
+        IProject project = projectManager.getProjectDetails(projectId);
+
+        // Identify the access type of the project : PUBLIC or PRIVATE
+        EProjectAccessibility projectAccess = project.getProjectAccess();
+
+        // If the project is PRIVATE, check if the user is allowed to access the
+        // information associated with the project
+        // If the project is PUBLIC, allow access to the information associated
+        // with the project
+        if (projectAccess == EProjectAccessibility.PRIVATE) {
+            boolean authorizedAccess = false;
+
+            if (projectSecurity.isProjectOwner(userId, projectId)) {
+                authorizedAccess = true;
+            } else {
+                List<String> collaboratorRoles = projectSecurity.getCollaboratorRoles(userId, projectId);
+                if (collaboratorRoles.contains(RoleNames.ROLE_PROJ_COLLABORATOR_EDITOR)
+                        || collaboratorRoles.contains(RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN)
+                        || collaboratorRoles.contains(RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR)) {
+                    authorizedAccess = true;
+                }
+            }
+
+            if (authorizedAccess) {
+                return networkVisualization(networkId, model);
+            }
+            return "public/forbidden";
+        } else {
+            return networkVisualization(networkId, model);
+        }
+    }
 
+    private String networkVisualization(String networkId, ModelMap model) throws QuadrigaStorageException {
+        INetwork network = networkManager.getNetwork(networkId);
+        if (network == null) {
+            return "auth/404";
+        }
+        ITransformedNetwork transformedNetwork = transformationManager.getTransformedNetwork(networkId);
+        String nwId = "\"" + networkId + "\"";
```

Comment:
why does the network id need to be surrounded by quotes?

Reply:
If we do not surround the network id with the quotes: String nwId = networkId; there is a reference error and the graph is not displayed:
![screen shot 2017-09-20 at 11 00 38 am](https://user-images.githubusercontent.com/23473237/30659574-0d5220f6-9df3-11e7-9688-07e34116d6b9.png)

Reply:
tiles-defs.xml contains the following definition:
	<definition name="auth/networks/visualize" extends="base.definition.rightnav">
		<put-attribute name="currentPage" value="networks" />
		<put-attribute name="title" value="Quadriga Networks Visualize" />
		<put-attribute name="right-navigation"
			value="/WEB-INF/views/navigation/nav-networkrightplane.jsp" />
		<put-attribute name="content"
			value="/WEB-INF/views/networks/networkvisual.jsp" />
	</definition>

networkvisual.jsp references the newtorkid (passed from NetworkListController.java) to invoke d3visualize function defined in (d3networks.js)

Reply:
If we surround the network id with quotes: String nwId = "\"" + networkId + "\"";
the graph is displayed without error.
![screen shot 2017-09-20 at 11 12 12 am](https://user-images.githubusercontent.com/23473237/30660344-4f15e322-9df5-11e7-8c24-b42636f64cac.png)

Reply:
That probably means that the javascript doesn't put quotations around it in the appropriate places. The javascript should be fixed rather than adding quotation marks here. Otherwise it would prevent the page to use the network id where only the id is required.